### PR TITLE
Guess config directories in install.sh

### PR DESCRIPTION
**Pull request by chrboe**
Tested only on Linux with Chromium due to time constraints (and because I don't have a Mac at hand 😜). Feel free to test on other platforms.

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -6,141 +6,142 @@ DIR="$( cd "$( dirname "$0" )" && pwd )"
...
-  fi
-  cp "$DIR/chrome-host.json" "$TARGET_DIR/$APP_NAME.json"
-else
-  if [ ! -f "$DIR/firefox-host.json" ]; then
-    echo "ERROR: '$DIR/firefox-host.json' is missing."
-    echo "If you are running './install.sh' from a release archive, please file a bug."
-    echo "If you are running './install.sh' from the source code, make sure to follow CONTRIBUTING.md on how to build first."
+if [ "$no_browser_found" = true ]; then
+    echo "No compatible browsers found."
+    echo "If  you do  actually have a  browser installed  which you  think should  be supported,"
```
Comment (maximbaz) [inline install.sh]: Is this a simulation of Word's "text align - justify" here? 😄

File: install.sh [hunk 2]
```diff
@@ -6,141 +6,142 @@ DIR="$( cd "$( dirname "$0" )" && pwd )"
...
-  echo "$OPERATING_SYSTEM is not supported"
-  exit 1
-  ;;
+    Linux)
+        HOST_FILE="$DIR/browserpass-linux64"
+        if [ "$(whoami)" == "root" ]; then
+            BROWSER_PATHS=( "Chrome:/etc/opt/chrome/native-messaging-hosts"
+            "Chromium:/etc/chromium/native-messaging-hosts"
+            "Firefox:/usr/lib/mozilla/native-messaging-hosts"
+            "Vivaldi:/etc/chromium/native-messaging-hosts" )
```
Comment (maximbaz) [inline install.sh]: Chromium and Vivaldi have the same folder, so Vivaldi is being falsely discovered as available. Do you think it is a copy-paste issue that nobody has discovered so far? 😄

**Comment by maximbaz** [comment]
So although I only have Chromium, your script discovered Chrome, Chromium and Firefox as available 🙂 

There is a good reason for that, I had them installed at some point, but although I deleted them, the configuration folders remained.

I'm concerned that we will be getting many new issues like "Hey, your install.sh says I have Firefox installed although I do not". As an idea, if user answers "no", maybe we can print a message, something like "If you feel this suggestion was a mistake and you don't have this browser installed, delete this folder: \<full-path\>"?

**Comment by maximbaz** [comment]
Also, how hard would it be to do the same for Powershell script, so we can fully close the #136? /cc @qbit

**Comment by qbit** [comment]
mm I guess it depends on the other browsers and what location they use to store the registry stuff. Shouldn't be much harder than determining the locations.

**Comment by maximbaz** [comment]
Is it possible to use the same approach as here, i.e. since all Powershell does is writing into registry as `hk{0}:\Software\BROWSER_NAME\NativeMessagingHosts`, just check if registry key `hk{0}:\Software\BROWSER_NAME` exists, and if yes, prompt "Detected BROWSER_NAME, install for this browser? [Y/n]"

File: install.sh (hunk 1 above)
Comment (chrboe) [reply install.sh]: Yep 😄 It really bothered me how the line ends were *almost* aligned, but not quite. Idk, it was a bit of an experiment. If you think it looks ugly I can still revert it 🤷‍♂️

File: install.sh (hunk 2 above)
Comment (chrboe) [reply install.sh]: Wow, I hadn't actually noticed that. It's literally been there [forever](https://github.com/dannyvankooten/browserpass/commit/9a313be3ad53ecdfb4bc53dd1c2543af5570038f#diff-3fbb47e318cd8802bd325e7da9aaabe8R31) though 😄 

I'll follow up with a fix sometime today.

**Comment by chrboe** [comment]
@maximbaz 

> I had them installed at some point, but although I deleted them, the configuration folders remained.

That's weird, shouldn't the package manager take care of deleting the config directory? 🤔
I guess the solution you propose would work just fine, but it kind of seems like a botch to me (from a user perspective). If we had an FAQ section I guess we could put an explanation there and print a link to it in `install.sh`.

Also, since I'm an Arch user myself, it kind of worries me how pacman just leaves stale config directories behind 😄 Do you happen to know whether or not there's a good reason for that?

Anyways, in the meantime I'll implement your proposed solution.

File: install.sh (hunk 1 above)
Comment (maximbaz) [reply install.sh]: It looks a little bit weird to me, let's revert 🙂

**Comment by maximbaz** [comment]
Yeah it makes sense indeed. 

I'm fine with both approaches, either print such warning directly from the script, or add this explanation to a README.md and refer to that section from the script. I'd choose depending on the size of the message.

**Comment by maximbaz** [comment]
Cool cool :+1: Are you planning on doing anything else here, or it is ready to be merged?

I'm planning on doing some final tests and merging all open PRs tomorrow in the afternoon.

**Comment by chrboe** [comment]
From my point of view I'm done here. The only thing I'm not 100% certain about is the global Vivaldi path. I looked around on the internet but couldn't find a definite source on whether or not `/etc/vivaldi` is actually correct.

It would be nice if someone who has Vivaldi installed could test installing with root (or even just check if the path is correct)

**Comment by maximbaz** [comment]
So I installed Vivaldi, and my experiments show that it doesn't read from `/etc` at all, neither from `/etc/vivaldi`, nor from `/etc/Vivaldi`, nor from `/etc/chromium`... It only reads from `$HOME/.config/vivaldi`.

**Comment by maximbaz** [comment]
Another test that I made:

- delete chromium entirely
- delete `/etc/chromium`
- install chromium
- check if `/etc/chromium` exists - it does not. So `sudo ./install.sh` doesn't detect that I have chromium installed.

So I don't like this approach of checking the existence of the config folders for detecting if a browser is available... Maybe we can do something else instead? Just a wild idea, maybe detect the existence of binaries in `$PATH`?

**Comment by chrboe** [comment]
Dang, that sucks... I guess *discovering* the browser would work using the `$PATH` method (or `whereis`?), but what would we do after that? Just create the config directory on behalf of the browser if it doesn't exist?

**Comment by maximbaz** [comment]
That's what we do today, actually :)

https://github.com/dannyvankooten/browserpass/blob/d5fc4fddf1d7a21455ab342e80d6472fe623b287/install.sh#L113-L114

Like I'm pretty sure I haven't had `/etc/chromium` at all until I started using browserpass.

I'm more concerted of how we implement the discovery mechanism, have you seen how many binaries the makefile tries just for Chrome and Chromium? :) I hope Firefox and others have less potential names :)

https://github.com/dannyvankooten/browserpass/blob/d5fc4fddf1d7a21455ab342e80d6472fe623b287/makefile#L2

**Comment by chrboe** [comment]
@maximbaz just wanted to give an update on this PR real quick. I'm pretty busy right now with university starting again next week, so I haven't had much time to work on browserpass. I will definitely come back to this patch very soon™️ though!

**Comment by maximbaz** [comment]
Sounds good :+1:

**Comment by maximbaz** [comment]
Closing due to https://github.com/dannyvankooten/browserpass/issues/136#issuecomment-353999726

**Pull request closed without merging**